### PR TITLE
Fix aliased type support in ImplicitRunInfo

### LINKED ISSUE #17305: assertion failed: `wildApprox` failed to remove uninstantiated B (should be a type error)
## Compiler version

3.3.0-RC3

## Minimized code

[Scastie link](https://scastie.scala-lang.org/nOHfRvjsS3eKkRggGhv2XQ); same code:

```scala
import zio.*
import zio.test.*
import zio.test.Assertion.*
import zio.Duration

enum TimeUnit derives CanEqual:
  case Millis
  case Seconds

final case class Perf(
  point: Int,
  time: Duration,
  series: String = "",
  displayUnit: TimeUnit = TimeUnit.Seconds
) {
  def timeToDisplayUnit: Long = displayUnit match
    case TimeUnit.Millis  => time.toMillis
    case TimeUnit.Seconds => time.toSeconds

}

  def assertLessThanBenches(longTimeValues: Chunk[Long])(tol: Double)(perfs: Chunk[Perf]): TestResult =
    longTimeValues
      .zip(perfs)
      .foldLeft(assertTrue(true))((acc, ii) => acc && assert(ii._2.displayUnit)(isLessThanEqualTo(ii._1 * (1.0d + tol))))
```

Note the `assertLessThanBenches` has what should be a type error; the correct code is:

```scala
      .foldLeft(assertTrue(true))((acc, ii) => acc && assert(ii._2.timeToDisplayUnit.toDouble)(isLessThanEqualTo(ii._1 * (1.0d + tol))))
```

## Output (click arrow to expand)

```
java.lang.AssertionError: assertion failed: `wildApprox` failed to remove uninstantiated B while compiling /tmp/scastie9219630149224372424/src/main/scala/main.scala
```

## See also

https://github.com/lampepfl/dotty/issues/14145

### PR DESCRIPTION
**Pull request by andrzejressel**
Fixes #17305

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/typer/Implicits.scala [hunk 1]
```diff
@@ -610,6 +610,8 @@ trait ImplicitRunInfo:
         else
           partSeen += t
           t.dealias match
+            case dealias if t ne dealias =>
```
Comment (odersky) [inline compiler/src/dotty/tools/dotc/typer/Implicits.scala:613]: I don't see any motivation for this code, and without it the change seems unsystematic. There is no wildApprox anywhere else in this function. Why introduce it here?

**Review by odersky — CHANGES_REQUESTED** [review]
The current fix like a patch for a problem introduced elsewhere. I think we need to focus on `wildApprox` instead.

File: compiler/src/dotty/tools/dotc/typer/Implicits.scala (hunk 1 above)
Comment (andrzejressel) [reply compiler/src/dotty/tools/dotc/typer/Implicits.scala:613]: I have no idea how to find a better way to put this check. The only thing I've discovered is that this type is not going through `wildApprox` when I guess it should.

I'll close this MR then.

**Comment by andrzejressel** [comment]
https://github.com/lampepfl/dotty/commit/bee299e792116f471f12dadf31360c7d208130b3 - reverting this also seems to work ... `(though we might
need to actually fix `typedArgs` one day if the same issue shows up in
other circumstances)` - I guess, I'll try to look for that

**Pull request closed without merging**